registry/cache/warming.go

```diff
@@ -151,7 +150,7 @@ func (w *Warmer) warm(ctx context.Context, logger log.Logger, id image.Name, cre
 	tags, err := client.Tags(ctx)
 	if err != nil {
 		if !strings.Contains(err.Error(), context.DeadlineExceeded.Error()) && !strings.Contains(err.Error(), "net/http: request canceled") {
-			logger.Log("err", errors.Wrap(err, "requesting tags"))
+			logger.Log("err", errors.Wrapf(err, "requesting tags for %v from %v using %v", id.CanonicalName(), creds.Hosts(), creds))
```

Comment:
Since we're using structured logging (e.g., the pair of `"err"` and the error), it's better to include the image name and credentials used as fields of the log message rather than as part of the error.

Reply:
I was thinking we could also have:
```go
logger = log.With(logger, "canonical_name", id.CanonicalName(), "creds", creds)
```
at the top of the method so that these details are added for all log statements.
Would this be preferable?

Reply:
That's a great idea

Reply:
OK. Done. And [example statement updated](https://github.com/weaveworks/flux/pull/898#issue-288221611).